src/PopIII/popiii.cpp

```diff
@@ -0,0 +1,431 @@
+//==============================================================================
+// TwoMomentRad - a radiation transport library for patch-based AMR codes
+// Copyright 2020 Benjamin Wibking.
+// Released under the MIT license. See LICENSE file included in the GitHub repo.
+//==============================================================================
+/// \file popiii.cpp
+/// \brief Defines a test problem for Pop III star formation.
+/// Author: Piyush Sharda (Leiden University, 2023)
+///
+#include <limits>
+#include <memory>
+#include <random>
+
+#include "AMReX.H"
+#include "AMReX_Arena.H"
+#include "AMReX_BC_TYPES.H"
+#include "AMReX_BLassert.H"
+#include "AMReX_Config.H"
+#include "AMReX_FabArrayUtility.H"
+#include "AMReX_MultiFab.H"
+#include "AMReX_ParallelDescriptor.H"
+#include "AMReX_ParmParse.H"
+#include "AMReX_Print.H"
+#include "AMReX_SPACE.H"
+#include "AMReX_TableData.H"
+
+#include "RadhydroSimulation.hpp"
+#include "SimulationData.hpp"
+#include "TurbDataReader.hpp"
+#include "hydro_system.hpp"
+#include "popiii.hpp"
+
+#include "actual_eos_data.H"
+#include "burn_type.H"
+#include "eos.H"
+#include "extern_parameters.H"
+#include "network.H"
+
+using amrex::Real;
+
+struct PopIII {
+};
+
+template <> struct HydroSystem_Traits<PopIII> {
+	static constexpr bool reconstruct_eint = false;
+};
+
+template <> struct Physics_Traits<PopIII> {
+	// cell-centred
+	static constexpr bool is_hydro_enabled = true;
+	static constexpr bool is_chemistry_enabled = false;	     // in the future, this could point to microphysics, and set to true
```

Comment:
We can get rid of this parameter, right? Everything is set with `numMassScalars`, plus the runtime parameters in the `.in` file, no?

Reply:
Yeah, we don't really use `is_chemistry_enabled`

Reply:
We should probably make a separate PR to get rid of this.

Reply:
True.